### ISSUE TITLE
Add option to sort translations by keys in resulting Strings files

### ISSUE THREAD

**Issue by Dschee** [Flinesoft/BartyCrouch#26]
Currently when a Strings file is saved the order of the translations isn't changed, new translations are simply added at the end of the list. There should be an option to sort them by keys which is especially useful for the `code` sub command when using structured keys like 'SETTINGS.USER_SECTION.LOGOUT'.

**Comment by kafejo** [comment]
I would also consider time sort option - it means that new strings will be added at the end. Because I often find myself searching through the localizable file to find a new key I've just added.

**Comment by Dschee** [comment]
@kafejo That should be the current behavior though, are you sure it's not the case for you?

**Comment by kafejo** [comment]
Actually I've examined it and current behaviour sorts it by key! Strange 😅
from our strings file:
```
… 
/* No advert present. */
"advert-detail.no-advert-error" = "The advert does not exist.";

/* Reissue action button title of advert peek preview */
"advert-preview.actions.sold" = "Reissue advert";

/* The user does not exist. */
"user-profile.no-user" = "The user does not exist.";

/* Camera */
"action-sheet.camera" = "Take photo";

/* Choose option */
"action-sheet.choose-option" = "Choose option";

/* Library */
"action-sheet.library" = "Photo library";

/* Will ship */
"adver.will-ship" = "Will ship";

/* Advert does not exist. */
"advert-administration.advert-does-not-exist" = "Advert does not exist.";

/* Restricted camera access */
"advert-administration.camera-access-restricted" = "Camera access for this user is restricted.";
… 
```
We use Bartycrouch run script in aggregated target
```
if [ $CONFIGURATION == Debug ]; then
if which bartycrouch > /dev/null; then
# Incrementally update all Storyboards/XIBs strings files
bartycrouch interfaces -p "$PROJECT_DIR"

# Add new keys to Localizable.strings files from NSLocalizedString in code
bartycrouch code -p "$PROJECT_DIR/maarked" -l "$PROJECT_DIR" -a
else
echo "warning: BartyCrouch not installed, download it from https://github.com/Flinesoft/BartyCrouch"
fi
fi
```

**Comment by Dschee** [comment]
I'm sorry, I still don't get it. What's sorted about those keys in your example? As far as I can see `advert-detail.no-advert-error` (lets call it A) appears first, then there's a `user-profile.no-user` key (B) and further down there is `advert-administration.advert-does-not-exist` (C).

If those keys were sorted then the order would need to be: C-A-B. But if the quoted file is the result of BartyCrouch then it's clearly not sorting by anything, instead it's just adding new ones as it should.

Btw: I think the best way to explain any odd behavior you wouldn't expect is to fork BartyCrouch, add a new test case that reproduces the unexpected behavior. This I would both understand it and also have a test that could make sure an implementation of the feature actually does the right thing. :)

**Comment by Dschee** [comment]
I just happened to come across this same issue in a project where PhraseApp is used. Gonna investigate this and report once I have results.

**Comment by Dschee** [comment]
@kafejo I have just added a simple test in fa9739d57f30063350e98ca143b9f5a5aecf15f8 to reproduce the issue we both have come across but without success. I expected the test to fail as we have both seen such a thing happen but it didn't – the new key was added at the end of the file.

I also copied the entire `Localizable.strings` over from the project where I had seen this issue – still the new key was added at the end, like expected. I think now that the test is setup you can easily alter the existing `Localizable.strings` file and/or the `SwiftExample3.swift` file(s) until you succeed in reproducing this issue. Once we know what the problem is we can fix it.

**Comment by kafejo** [comment]
Nice. I will take a look at it.

**Comment by Dschee** [comment]
@kafejo I just found the issue with the reordering of existing keys and fixed it – just update to version 3.3.5 to see for yourself. I hope that fixes your issue! :)

**Comment by kafejo** [comment]
Cool! Thanks! 🎉

**Comment by Dschee** [comment]
I just implemented this feature and released a new version with it – use the `-s` (sort-by-keys) option to sort your files by their keys starting with version 3.5.0. 🎉

**Comment by Dschee** [comment]
Please note that I just released version 3.5.1 which changes the behavior of the sort-by-keys option slightly: It sorts entries with empty translations to the end so untranslated keys can be easily found as expected. Another run of BartyCrouch after translating the keys will sort them correctly. I think this is gonna make this option much more useful in practice! :)
